holidays/countries/israel.py

```diff
@@ -9,96 +9,159 @@
 #  Website: https://github.com/dr-prodigy/python-holidays
 #  License: MIT (see LICENSE file)
 
-from datetime import date
 from datetime import timedelta as td
 
 from holidays.calendars import _HebrewLunisolar
-from holidays.holiday_base import HolidayBase
-
-
-class Israel(HolidayBase):
+from holidays.constants import OPTIONAL, PUBLIC, SCHOOL
+from holidays.observed_holiday_base import (
+    ObservedHolidayBase,
+    MON_TO_NEXT_TUE,
+    THU_TO_PREV_WED,
+    FRI_TO_PREV_WED,
+    FRI_TO_PREV_THU,
+    SAT_TO_PREV_THU,
+    SAT_TO_NEXT_SUN,
+    SUN_TO_NEXT_MON,
+)
+
+
+class Israel(ObservedHolidayBase):
     country = "IL"
+    observed_label = "%s (Observed)"
+    supported_categories = {OPTIONAL, PUBLIC, SCHOOL}
 
-    def _add_with_eve(self, name: str, dt: date) -> None:
-        self._add_holiday(f"{name} - Eve", dt + td(days=-1))
-        self._add_holiday(name, dt)
+    def __init__(self, *args, **kwargs):
+        kwargs.setdefault("observed_rule", FRI_TO_PREV_THU + SAT_TO_PREV_THU)
+        super().__init__(*args, **kwargs)
 
-    def _populate(self, year):
-        if year <= 1947:
+    def _populate_public_holidays(self):
+        if self._year <= 1947:
             return None
 
-        if year >= 2101:
+        if self._year >= 2101:
             raise NotImplementedError
 
-        super()._populate(year)
+        # Rosh Hashanah
+        rosh_hashanah_dt = _HebrewLunisolar.hebrew_holiday_date(self._year, "ROSH_HASHANAH")
+        name = "Rosh Hashanah"
+        self._add_holiday(name, rosh_hashanah_dt)
+        self._add_holiday(name, rosh_hashanah_dt + td(days=+1))
 
-        # Passover
-        passover_dt = _HebrewLunisolar.hebrew_holiday_date(year, "PASSOVER")
-        self._add_with_eve("Passover I", passover_dt)
-        for offset in range(1, 5):
-            self._add_holiday("Passover - Chol HaMoed", passover_dt + td(days=offset))
-        self._add_with_eve("Passover VII", passover_dt + td(days=+6))
+        # Yom Kippur
+        yom_kippur_dt = _HebrewLunisolar.hebrew_holiday_date(self._year, "YOM_KIPPUR")
+        self._add_holiday("Yom Kippur", yom_kippur_dt)
```

Comment:
Could you inline values used only once?
```suggestion
        self._add_holiday("Yom Kippur", self._get_holiday("YOM_KIPPUR"))
```